src/cryptenroll/cryptenroll-tpm2.c

```diff
@@ -65,7 +65,7 @@ int enroll_tpm2(struct crypt_device *cd,
         _cleanup_(erase_and_freep) char *base64_encoded = NULL;
         size_t secret_size, secret2_size, blob_size, hash_size;
         _cleanup_free_ void *blob = NULL, *hash = NULL;
-        uint16_t pcr_bank;
+        uint16_t pcr_bank, primary_alg;
```

Comment:
What does 'primary' stand for here and in all other references to the key algorithm?

Reply:
It's TPM2 parlance. The "primary" key is the key you derive from the TPM2 seed, and that all the keys you actually intend to generate/download/upload/use are encrypted with. It's an asymmetric key that the TPM2 will encrypt your keys when you download them out of the TPM2, or that it will use to decrypt them again when you upload them again.

It's really a core concept of TPM2 I guess, so I think we don't really need much of an explanation here. But there actually is one nevertheless in our sources, see the big comment in tpm2_unseal(), which gives an overview how this all fits together.

Reply:
thanks, makes sense